cmd/build/v2/tagger.go

```diff
@@ -103,80 +108,33 @@ func (it imageTagger) getImageReferencesForTag(manifestName, svcToBuildName, tag
 	}
 
-	imageReferences = append(imageReferences, i.getImageReferencesForTag(manifestName, svcToBuildName, model.OktetoDefaultImageTag)...)
+	// The default tag (okteto) should be considered only with the dev registry. It should never be used with the global
+	// registry
+	sanitizedName := format.ResourceK8sMetaString(manifestName)
+	imageReferences = append(imageReferences, useReferenceTemplate(constants.DevRegistry, sanitizedName, svcToBuildName, model.OktetoDefaultImageTag))
 
 	return imageReferences
 }
 
-// imageTaggerWithVolumes represent an imageTaggerInterface with an reference tag with volume mounts
-type imagerTaggerWithVolumes struct {
-	cfg                  oktetoBuilderConfigInterface
-	smartBuildController smartBuildController
-}
-
-// newImageWithVolumesTagger returns a new image tagger
-func newImageWithVolumesTagger(cfg oktetoBuilderConfigInterface, sbc smartBuildController) imagerTaggerWithVolumes {
-	return imagerTaggerWithVolumes{
-		cfg:                  cfg,
-		smartBuildController: sbc,
-	}
-}
-
-// getServiceImageReference returns the full image tag for the build
-func (i imagerTaggerWithVolumes) getServiceImageReference(manifestName, svcName string, _ *build.Info, buildHash string) string {
-
-	targetRegistry := constants.DevRegistry
-	tag := ""
-	if i.cfg.HasGlobalAccess() && i.smartBuildController.IsEnabled() {
-		if i.cfg.IsCleanProject() {
-			targetRegistry = constants.GlobalRegistry
-		}
-		tag = buildHash
-	}
-	sanitizedName := format.ResourceK8sMetaString(manifestName)
-	if tag != "" {
-		return useReferenceTemplateWithVolumes(targetRegistry, sanitizedName, svcName, tag)
-	}
-	return useReferenceTemplate(targetRegistry, sanitizedName, svcName, model.OktetoImageTagWithVolumes)
-}
-
-// getImageReferencesForTag returns all the possible images that can be built from a commit hash
-func (i imagerTaggerWithVolumes) getImageReferencesForTag(manifestName, svcToBuildName, tag string) []string {
-	if tag == "" {
-		return []string{}
-	}
-	// manifestName can be not sanitized when option name is used at deploy
+// getImageReferencesForDeploy returns the list of images references for a service when deploying it. In case of deploy,
+// we only have to check if the image is present with the okteto tag. We don't check anything related to the hash
+func (imageTagger) getImageReferencesForDeploy(manifestName, svcToBuildName string) []string {
+	var imageReferences []string
 	sanitizedName := format.ResourceK8sMetaString(manifestName)
-	tagsToCheck := []string{}
-	for _, targetRegistry := range getTargetRegistries(i.cfg.IsOkteto()) {
-		tagsToCheck = append(tagsToCheck, useReferenceTemplateWithVolumes(targetRegistry, sanitizedName, svcToBuildName, tag))
-	}
-	return tagsToCheck
-}
+	imageReferences = append(imageReferences, useReferenceTemplate(constants.DevRegistry, sanitizedName, svcToBuildName, model.OktetoDefaultImageTag))
 
```

Comment:
NIT: For me it's easier to read it like this but feel free to leave it as it is
```suggestion
	sanitizedName := format.ResourceK8sMetaString(manifestName)
	imageReferences := []string{useReferenceTemplate(constants.DevRegistry, sanitizedName, svcToBuildName, model.OktetoDefaultImageTag)}
	
	return imageReferences
```

Reply:
Changed!